Downloading `ZIM files` are not working in `Play Store Variant`

**Describe the bug**

Downloading `ZIM files` are not working in `Play Store Variant`.

**Expected behaviour**

It should download the `ZIM files` as it is downloading in `All other variant` .

**Steps to reproduce the behavior:**
* Install `Play Store` variant.
* Go to download screen.
* Click on any `ZIM file` to download.

**Screenshots**


https://user-images.githubusercontent.com/34593983/231685395-3220e5bf-f61c-4d43-9680-34f866af14f0.mp4


<br>

**Environment**
- Version of Kiwix Android : 3.6.0
- Device : Redmi Note 9 (Android 12) , Android 11 Emulator.

**Logs**
```
Caused by: org.simpleframework.xml.core.PersistenceException: Constructor not matched for class org.kiwix.kiwixmobile.core.entity.MetaLinkNetworkEntity$FileElement
```

```
Caused by: org.simpleframework.xml.core.PersistenceException: Constructor not matched for class org.kiwix.kiwixmobile.core.entity.MetaLinkNetworkEntity$Pieces
```

--- a/app/proguard-rules.pro
+++ b/app/proguard-rules.pro
@@ -42,3 +42,10 @@
 -keepclassmembers class * {
     @org.simpleframework.xml.* *;
 }
+
+## keep everything in MetaLinkNetworkEntity.kt
+
+-keepnames class org.kiwix.kiwixmobile.core.entity.MetaLinkNetworkEntity$*
+-keepclassmembers class org.kiwix.kiwixmobile.core.entity.MetaLinkNetworkEntity$* {
+    <init>(...);
+}

